File: gw_dangan_back/src/main/java/com/example/GwDanganApp/models/tasks/Task.java
package com.example.GwDanganApp.models.tasks;

import com.example.GwDanganApp.models.users.User;
import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;

import jakarta.persistence.Table;
import jakarta.validation.constraints.NotBlank;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "tasks")
@Getter
@Setter
@NoArgsConstructor
public class Task {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @NotBlank(message = "Task name is required")
    private String name;
    private String description;
    
    @Enumerated(EnumType.STRING)
    private Status status;
    
    @Enumerated(EnumType.STRING)
    private Priority priority;
    
    private Boolean isCompleted;
    
    @ManyToOne
    @JoinColumn(name = "author_id")
    @JsonIgnore
    private User author;

    public Task(String name, String description, User author) {
        this.name = name;
        this.description = description;
        this.author = author;
        this.status = Status.NOT_STARTED;
        this.priority = Priority.LOW;
        this.isCompleted = false;
    }

    public String getAuthorId() {
        return author.getFirebaseUid();
    }

    public String getAuthorName() {
        return author.getName();
    }
}

File: gw_dangan_front/lib/models/tasks/task.dart
class Task {
  final int id;
  final String name;
  final String description;
  final String status;
  final String priority;
  final bool isCompleted;

  Task({
    required this.id,
    required this.name,
    required this.description,
    required this.status,
    required this.priority,
    required this.isCompleted,
  });

  factory Task.fromJson(Map<String, dynamic> json) {
    return Task(
      id: json['id'],
      name: json['name'],
      description: json['description'],
      status: json['status'],
      priority: json['priority'],
      isCompleted: json['isCompleted'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'description': description,
      'status': status,
      'priority': priority,
      'isCompleted': isCompleted,
    };
  }

  Task copyWith({
    int? id,
    String? name,
    String? description,
    String? status,
    String? priority,
    bool? isCompleted,
  }) {
    return Task(
      id: id ?? this.id,
      name: name ?? this.name,
      description: description ?? this.description,
      status: status ?? this.status,
      priority: priority ?? this.priority,
      isCompleted: isCompleted ?? this.isCompleted,
    );
  }
}
